plugin/src/main/groovy/com/btkelly/gnag/reporters/AndroidLintViolationDetector.groovy

```diff
@@ -52,15 +52,39 @@ class AndroidLintReporter implements Reporter {
     }
 
     @Override
-    boolean hasErrors() {
+    List<Violation> getDetectedViolations() {
         GPathResult xml = new XmlSlurper().parseText(reportFile().text)
-        int numErrors = xml.issue.count { severityEnabled(it.@severity.text()) }
-        println "Android Lint report executed, found " + numErrors + " errors."
-        return numErrors != 0
+
+        final List<Violation> result = new ArrayList<>()
+        
+        xml.issue.findAll { severityEnabled((String) it.@severity.text()) }
+                .each { violation ->
+                        final Integer lineNumber;
+            
+                        try {
+                            lineNumber = violation.location.@line.toInteger()
+                        } catch (final NumberFormatException e) {
+                            System.out.println("Error reading line number from Android Lint violations.");
+                            e.printStackTrace();
+                            lineNumber = null
+                        }
+
+                        result.add(new Violation(
+                                sanitize((String) violation.@id.text()),
```

Comment:
Is there a reason the sanitize function isn't just inside the Violation constructor?